File: downtimeC/RestartWheel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using System.Collections;
using System.Diagnostics;

using System.Text.RegularExpressions;
using HL7;
using FunctionalCSharp;

namespace downtimeC
{
    public class OrderNumberForComobBox
    {
        public readonly string AlphaVersion;
        public readonly string IntVersion;

        public OrderNumberForComobBox(string AlphaVersion, string IntVersion)
        {
            this.IntVersion = IntVersion;
            this.AlphaVersion = AlphaVersion;
        }

        public override string ToString()
        {
            return AlphaVersion;
        }
    }

    public partial class RestartWheel : Form
    {
        protected RestartWheel()
        {
            InitializeComponent();
        }

        readonly GetSqlServer getSqlServer;
        public RestartWheel(GetSqlServer getSqlServer)
        {
            InitializeComponent();
            this.getSqlServer = getSqlServer;
        }

        string TDate;
        public void submitneworder(string ordernumber)
        {
            getSqlServer.ExecuteNonQuery("truncate table [ordered]");


            TDate = DateTimeFunctions.datetomysql(System.DateTime.Now.ToString());

            //..........//////use to reinstate alpha-numeric(also un-comment line items 540-547 in orderentry)\\\\\\\\.........


            getSqlServer.ExecuteNonQuery("insert into ordernumber (OrderLast,Ordernumber, Date) VALUES (0," + ordernumber + ", '" + TDate + "');");
        }


        private void Button1_Click(System.Object sender, System.EventArgs e)
        {
            getIntOrderNumber().forEach(ordernumber =>
            {
                if (ComboBoxNewOrderNumber.SelectedIndex.ToString().Length > 1)
                {
                    Interaction.MsgBox("Ordernumber has not been selected.  Please select Ordernubmer", MsgBoxStyle.OkOnly, "MsgBox");
                    return;
                }

                var response = Interaction.MsgBox("Are you Sure you Want to Clear ALL DATA????", MsgBoxStyle.YesNo, "MsgBox");
                if (response == MsgBoxResult.Yes)
                {
                    getSqlServer.ExecuteNonQuery("truncate table ordernumber");
                    submitneworder(ordernumber);
                    Interaction.MsgBox("Old Data Has Been Cleard and Ordernumber Has Been Reset.", MsgBoxStyle.OkOnly, "MsgBox");
                }
            });
        }

        private void Buttondelettrack_Click(System.Object sender, System.EventArgs e)
        {
            var response = Interaction.MsgBox("Are you Sure you Want to reset tracking?", MsgBoxStyle.YesNo, "MsgBox");
            if (response == MsgBoxResult.Yes)
            {
                getSqlServer.ExecuteNonQuery("truncate table dttracking");
            }
        }

        public string date2ordernumber(string dates)
        {
            string ordend = dates;

            System.Text.RegularExpressions.Match dateend = Regex.Match(ordend, "([0-9]+)/([0-9]+)/([0-9]+)");
            string endmonth = dateend.Groups[1].Value;
            string endday = dateend.Groups[2].Value;
            string endyear = dateend.Groups[3].Value;
            while (endday.Length < 2)
            {
                endday = "0" + endday;
            }



            int endmnth = int.Parse(endyear) - 2004;
            endmnth = endmnth * 12;


            string monthend = Convert.ToString(Convert.ToInt32(endmonth) + Convert.ToInt32(endmnth));

            if (int.Parse(monthend) > 99)
            {
                int newmonthstart = int.Parse(Strings.Left(monthend, 2));
                string newmonthend = Strings.Right(monthend, 1);
                newmonthstart = newmonthstart + 55;
                monthend = Microsoft.VisualBasic.Strings.Chr(newmonthstart) + newmonthend;
            }
            string monthday = monthend + endday;
            return (monthday);

        }


        private void restartwheel_Load(System.Object sender, System.EventArgs e)
        {
            //load combobox and Dictionary
            var dr = getSqlServer.FilledRow("select * from NumberWheel;");

            string datanumber = dr["OrderNumberAlpha"].ToString();
            string lownumber = dr["OrderNumberNumer"].ToString();
            //int n = Convert.ToInt32(lownumber);


            //while (n < 8001)
            //{
            //    string nonalphaordernumber = (date2ordernumber(System.DateTime.Now.ToString()) + n.ToString());
            //    ComboBoxNewOrderNumber.Items.Add(
            //        new OrderNumberForComobBox(nonalphaordernumber, n.ToString()));// nonalphaordernumber);

            //    n = n + 500;
            //}

            string ordernums = Strings.Right(datanumber, 3);
            char letters = Strings.Chr(int.Parse(Strings.Left(datanumber, 2)));

            string alphanum = date2ordernumber(System.DateTime.Now.ToString()) + letters + ordernums;

            ComboBoxNewOrderNumber.Items.Add(
                new OrderNumberForComobBox(alphanum, datanumber));
        }

        private void resetOrderNumber(string ordernumber)
        {
            getSqlServer.ExecuteNonQuery("truncate table ordernumber");

            submitneworder(ordernumber);

            Interaction.MsgBox("Order Number Has Been Reset", MsgBoxStyle.OkOnly, "MsgBox");
        }

        private void ButtonNumberReset_Click(System.Object sender, System.EventArgs e)
        {
            getIntOrderNumber().forEach(ordernumber =>
            {
                Option<DataRow> drOption = getSqlServer.FilledRowOption("SELECT TOP 1 Date FROM ordernumber ORDER BY Date DESC;");

                var reset = drOption.map(dr =>
                {
                    if (DateTime.Now.Date == dr.Field<DateTime>("Date"))
                    {
                        var response = Interaction.MsgBox(
                            "Order Number has already been reset for today.  Would you like to reset it again?"
                             , MsgBoxStyle.YesNo, "MsgBox");

                        if (response == MsgBoxResult.No) return false;
                    }

                    return true;
                });

                if (reset.getOrElse(() => true)) resetOrderNumber(ordernumber);
            });
        }

        private void ComboBoxNewOrderNumber_KeyDown(object sender, System.Windows.Forms.KeyEventArgs e)
        {
            e.SuppressKeyPress = true;
        }

        /// <summary>
        /// Get the version of the ordernumber that represents the CHAR in the number as Int
        /// </summary>
        /// <returns></returns>
        private Option<string> getIntOrderNumber()
        {
            if (ComboBoxNewOrderNumber.SelectedIndex.ToString().Length > 1)
            {
                Interaction.MsgBox("Ordernumber has not been selected.  Please select Ordernubmer", MsgBoxStyle.OkOnly, "MsgBox");
                return Option.None<string>();
            }

            return Option.Some(((OrderNumberForComobBox)ComboBoxNewOrderNumber.SelectedItem).IntVersion);
        }

    }

}

File: downtime/restartwheel.vb
Imports MySql.Data.MySqlClient
Imports System.Text.RegularExpressions
Public Class RestartWheel

    Dim monthday As String = ""
    Dim ordernumber As String = ""
    Dim TDate As String = ""
    Dim dict As New Dictionary(Of String, String)


    Public Shared Sub number()
        Dim myorder As New RestartWheel
        Application.Run(myorder)

    End Sub
    Sub Deleteoldinfo()



        Dim comm1 As New MySqlCommand("truncate table dtdb1.ordernumber", New MySqlConnection("server=lis-s22104-db1;uid=dniggli;pwd=vvo084;"))

        comm1.Connection.Open()
        comm1.ExecuteNonQuery()
        comm1.Connection.Close()

    End Sub


    Sub Deleteoldtrackinfo()

        Dim comm As New MySqlCommand("truncate table dtdb1.dttracking", New MySqlConnection("server=lis-s22104-db1;uid=dniggli;pwd=vvo084;"))

        comm.Connection.Open()
        comm.ExecuteNonQuery()
        comm.Connection.Close()

    End Sub


    Sub submitneworder()

        Dim comm2 As New MySqlCommand("truncate table dtdb1.Table1", New MySqlConnection("server=lis-s22104-db1;uid=dniggli;pwd=vvo084;"))

        comm2.Connection.Open()
        comm2.ExecuteNonQuery()
        comm2.Connection.Close()



        TDate = DateTimeFunctions.datetomysql(Date.Now)

        '..........//////use to reinstate alpha-neumaric(also un-comment line items 540-547 in orderentry)\\\\\\\\.........


        Dim comm As New MySqlCommand("insert into dtdb1.ordernumber (Ordernumber, Date) VALUES ('" & ordernumber & "', '" & TDate & "');", New MySqlConnection("server=lis-s22104-db1;uid=dniggli;pwd=vvo084;"))

        'Dim comm As New MySqlCommand("insert into dtdb1.ordernumber (Ordernumber, Date) VALUES ('5000', '" & TDate & "');", New MySqlConnection("server=lis-s22104-db1;uid=dniggli;pwd=vvo084;"))
        comm.Connection.Open()
        comm.ExecuteNonQuery()
        comm.Connection.Close()

    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click



        If ComboBoxNewOrderNumber.SelectedIndex.ToString.Length > 1 Then
            Dim msg1 As String
            Dim title1 As String
            Dim style1 As MsgBoxStyle
            Dim response1 As MsgBoxResult

            msg1 = "Ordernumber has not been selected.  Please select Ordernubmer"   ' Define message.
            style1 = MsgBoxStyle.OkOnly
            title1 = "MsgBox"   ' Define title.
            ' Display message.
            response1 = MsgBox(msg1, style1, title1)
            If response1 = MsgBoxResult.Ok Then
                Exit Sub
            End If

        End If


        Dim msg As String
        Dim title As String
        Dim style As MsgBoxStyle
        Dim response As MsgBoxResult

        msg = "Are you Sure you Want to Clear ALL DATA????"   ' Define message.
        style = MsgBoxStyle.YesNo
        title = "MsgBox"   ' Define title.
        ' Display message.
        response = MsgBox(msg, style, title)
        If response = MsgBoxResult.Yes Then
            Deleteoldinfo()
            submitneworder()

        End If
        If response = MsgBoxResult.No Then
            Exit Sub
        End If



        msg = "Old Data Has Been Cleard and Ordernumber Has Been Reset."   ' Define message.
        style = MsgBoxStyle.OkOnly
        title = "MsgBox"   ' Define title.
        ' Display message.
        response = MsgBox(msg, style, title)
        If response = MsgBoxResult.Ok Then

        End If


    End Sub

    Private Sub Buttondelettrack_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Buttondelettrack.Click
        Dim msg As String
        Dim title As String
        Dim style As MsgBoxStyle
        Dim response As MsgBoxResult

        msg = "Are you Sure you Want to reset tracking?"   ' Define message.
        style = MsgBoxStyle.YesNo
        title = "MsgBox"   ' Define title.
        ' Display message.
        response = MsgBox(msg, style, title)
        If response = MsgBoxResult.Yes Then
            Deleteoldtrackinfo()

        End If
        If response = MsgBoxResult.No Then Exit Sub

    End Sub

    Function date2ordernumber(ByVal dates As String)
        Dim ordend As String = dates

        Dim dateend As System.Text.RegularExpressions.Match = Regex.Match(ordend, "([0-9]+)/([0-9]+)/([0-9]+)")
        Dim endmonth As String = dateend.Groups(1).Value
        Dim endday As String = dateend.Groups(2).Value
        Dim endyear As String = dateend.Groups(3).Value
        While endday.Length < 2
            endday = "0" & endday
        End While



        Dim endmnth As Integer = endyear + -2004
        endmnth = endmnth * 12


        Dim monthend As String = CStr(CInt(endmonth) + CInt(endmnth))

        If monthend > 99 Then
            Dim newmonthstart As String = Microsoft.VisualBasic.Left(monthend, 2)
            Dim newmonthend As String = Microsoft.VisualBasic.Right(monthend, 1)
            newmonthstart = newmonthstart + 55
            monthend = Chr(newmonthstart) + newmonthend
        End If
        Dim monthday As String = monthend + endday
        Return (monthday)

    End Function

    Private Sub restartwheel_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load








        '.......................///////////// Create Alpha-neumaric OrderNumber \\\\\\\\\\\\\\\\\........................



        Dim DatabaseNumber As New MySqlDataAdapter("select * from dtdb1.NumberWheel;", "server=lis-s22104-db1;uid=dniggli;pwd=vvo084;")

        Dim dt As New DataTable
        DatabaseNumber.Fill(dt)

        Dim dr As DataRow = dt.Rows(0)

        Dim datanumber As String = dr("OrderNumberAlpha").ToString
        Dim lownumber As String = dr("OrderNumberNumer").ToString




        Dim n As Integer = Convert.ToInt32(lownumber)

        Do While n < 8001
            Dim nonalphaordernumber As String = (date2ordernumber(Date.Now) + n.ToString)
            ComboBoxNewOrderNumber.Items.Add(nonalphaordernumber)

            dict.Add(nonalphaordernumber, n.ToString)

            n = n + 500


        Loop

        Dim ordernums As String = Microsoft.VisualBasic.Right(datanumber, 3)
        Dim letters As String = Microsoft.VisualBasic.Left(datanumber, 2)
        Dim alphanum As String = date2ordernumber(Date.Now) + Chr(letters) + ordernums
        Dim ordnumend As String = letters + ordernums



        ComboBoxNewOrderNumber.Items.Add(alphanum)

        dict.Add(alphanum, ordnumend)






    End Sub

    Private Sub ButtonNumberReset_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ButtonNumberReset.Click


        If ComboBoxNewOrderNumber.SelectedIndex.ToString.Length > 1 Then
            Dim msg1 As String
            Dim title1 As String
            Dim style1 As MsgBoxStyle
            Dim response1 As MsgBoxResult

            msg1 = "Ordernumber has not been selected.  Please select Ordernubmer"   ' Define message.
            style1 = MsgBoxStyle.OkOnly
            title1 = "MsgBox"   ' Define title.
            ' Display message.
            response1 = MsgBox(msg1, style1, title1)
            If response1 = MsgBoxResult.Ok Then
                Exit Sub
            End If

        End If

        Dim ch As New MySqlDataAdapter("SELECT Date FROM dtdb1.ordernumber order BY Date DESC lIMIT 1;", "server=lis-s22104-db1;uid=dniggli;pwd=vvo084;")
        Dim dtble As New DataTable
        ch.Fill(dtble)

        Dim q As DataRow = dtble.Rows(0)
        Dim datestring As Date = q("Date")

        Dim checkdate As Date = Date.Now.Date


        If Not checkdate = datestring Then

            Dim comm1 As New MySqlCommand("truncate table dtdb1.ordernumber", New MySqlConnection("server=lis-s22104-db1;uid=dniggli;pwd=vvo084;"))

            comm1.Connection.Open()
            comm1.ExecuteNonQuery()
            comm1.Connection.Close()

            submitneworder()

            Dim msg1 As String
            Dim title1 As String
            Dim style1 As MsgBoxStyle
            Dim response1 As MsgBoxResult

            msg1 = "Order Number Has Been Reset"   ' Define message.
            style1 = MsgBoxStyle.OkOnly
            title1 = "MsgBox"   ' Define title.
            ' Display message.
            response1 = MsgBox(msg1, style1, title1)



        Else
            Dim msg As String
            Dim title As String
            Dim style As MsgBoxStyle
            Dim response As MsgBoxResult

            msg = "Order Number has already been reset for today.  Would you like to reset it again?"   ' Define message.
            style = MsgBoxStyle.YesNo
            title = "MsgBox"   ' Define title.
            ' Display message.
            response = MsgBox(msg, style, title)
            If response = MsgBoxResult.No Then

            ElseIf response = MsgBoxResult.Yes Then

                Dim comm1 As New MySqlCommand("truncate table dtdb1.ordernumber", New MySqlConnection("server=lis-s22104-db1;uid=dniggli;pwd=vvo084;"))

                comm1.Connection.Open()
                comm1.ExecuteNonQuery()
                comm1.Connection.Close()

                submitneworder()

                Dim msg1 As String
                Dim title1 As String
                Dim style1 As MsgBoxStyle
                Dim response1 As MsgBoxResult

                msg1 = "Order Number Has Been Reset"   ' Define message.
                style1 = MsgBoxStyle.OkOnly
                title1 = "MsgBox"   ' Define title.
                ' Display message.
                response1 = MsgBox(msg1, style1, title1)
            End If




        End If


    End Sub

    Private Sub ComboBoxNewOrderNumber_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles ComboBoxNewOrderNumber.KeyDown
        e.SuppressKeyPress = True
    End Sub

    Private Sub ComboBoxNewOrderNumber_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ComboBoxNewOrderNumber.SelectedIndexChanged

        ordernumber = dict(ComboBoxNewOrderNumber.SelectedItem)

    End Sub
End Class
